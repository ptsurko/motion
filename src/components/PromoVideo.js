import React from 'react';
import Types from 'prop-types';
import 'url-search-params-polyfill';
import { Link } from 'react-router-dom';

const PromoVideo = (props) => {
  const roomId = new URLSearchParams(props.location.search).get('roomId');

  return (
    <main className="promoVideo-container">
      <div className="promoVideo-overlay" />
      <iframe
        allowFullScreen
        className="promoVideo"
        frameBorder="0"
        src="https://player.vimeo.com/video/224553869?title=0&byline=0&portrait=0"
        title="Satori Motion Video"
      />
      <Link
        className="link--jumpIn"
        to={{ pathname: roomId ? `/room/${roomId}` : '/', search: 'currentState=main' }}>
        <button className="btn btn--primary">
          Jump in
        </button>
      </Link>
    </main>
  );
};

PromoVideo.propTypes = {
  location: Types.shape({
    search: Types.string.isRequired,
  }).isRequired,
};

export default PromoVideo;
