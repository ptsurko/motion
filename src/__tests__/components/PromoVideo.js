import React from 'react';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router-dom';
import PromoVideo from 'components/PromoVideo';

it('renders PromoVideo component correctly with roomId', () => {
  const tree = renderer.create(
    <StaticRouter context={{}}><PromoVideo location={{ search: '?roomId=abc' }} /></StaticRouter>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders PromoVideo component correctly without roomId', () => {
  const tree = renderer.create(
    <StaticRouter context={{}}><PromoVideo location={{ search: '' }} /></StaticRouter>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
