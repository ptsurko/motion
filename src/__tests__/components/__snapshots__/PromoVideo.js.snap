// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PromoVideo component correctly with roomId 1`] = `
<main
  className="promoVideo-container"
>
  <div
    className="promoVideo-overlay"
  />
  <iframe
    allowFullScreen={true}
    className="promoVideo"
    frameBorder="0"
    src="https://player.vimeo.com/video/222914713?title=0&byline=0&portrait=0"
    title="Satori Motion Video"
  />
  <a
    className="link--jumpIn"
    href="/room/abc?currentState=main"
    onClick={[Function]}
  >
    <button
      className="btn btn--primary"
    >
      Jump in
    </button>
  </a>
</main>
`;

exports[`renders PromoVideo component correctly without roomId 1`] = `
<main
  className="promoVideo-container"
>
  <div
    className="promoVideo-overlay"
  />
  <iframe
    allowFullScreen={true}
    className="promoVideo"
    frameBorder="0"
    src="https://player.vimeo.com/video/222914713?title=0&byline=0&portrait=0"
    title="Satori Motion Video"
  />
  <a
    className="link--jumpIn"
    href="/?currentState=main"
    onClick={[Function]}
  >
    <button
      className="btn btn--primary"
    >
      Jump in
    </button>
  </a>
</main>
`;
